FerramAerospaceResearch/FARAeroComponents/FARAeroSection.cs

```diff
@@ -336,12 +337,15 @@ public void UpdateSimulationContext(Vector4 worldVel, FARCenterQuery center, flo
 
             public Vector3 LocalVelocity(PartData pd)
             {
+                if (pd.aeroModule.part == null || pd.aeroModule.part.partTransform == null)
+                    return Vector3.zero;
```

Comment:
https://github.com/dkavolis/Ferram-Aerospace-Research/blob/593cf15f27bbd95f3a602c690cb795d5d956d53c/FerramAerospaceResearch/FARAeroComponents/FARAeroSection.cs#L340-L341
https://github.com/dkavolis/Ferram-Aerospace-Research/blob/593cf15f27bbd95f3a602c690cb795d5d956d53c/FerramAerospaceResearch/FARAeroComponents/FARAeroSection.cs#L347-L348

Rather than doing the same null check twice, do it once and skip the iteration if it is not satisfied. There is no need to waste computation on invalid parts.

Reply:
In this circumstance, should the non-simulation path also skip? It would be easiest just to lift these checks into CalculateAeroForces.

Reply:
In flight calculation mode, the null checks are not needed and I would like it to stay that way. Unity `Object` overrides equality operators with a call to native code so avoiding them in methods that are called frequently helps with performance.

Reply:
If this is performance critical (I'm still not sure that the simulation path is?), then I'm unhappy introducing another vcall if it can possibly be avoided. Another option would be to skip if the local velocity is exactly zero. In that case, the applied forces would then also be zero, so it shouldn't change the net result.

Reply:
MechJeb calls simulation from `FixedUpdate`. If you think checking if local velocity, or rather its `sqrMagnitude`, is zero is better than additional interface method call, then go ahead and do that. There's an extension method `FARMathUtil.NearlyEqual` for approximate floating point equality, negligible velocities could also be discarded this way.